File: app/src/main/java/com/proxy/shadowsocksr/impl/crypto/crypto/AbsCrypto.java
package com.proxy.shadowsocksr.impl.crypto.crypto;

/*
 * If you need add new Crypto, just extends this class, and change CryptoManager class.
 */
public abstract class AbsCrypto
{
    public String cryptMethod;
    public byte[] key;

    /**
     * @param cryptMethod Crypt method name,should lower case.
     * @param key         crypt key
     */
    public AbsCrypto(String cryptMethod, byte[] key)
    {
        this.cryptMethod = cryptMethod;
        this.key = key;
    }

    public abstract void updateEncryptIV(byte[] iv);//If cipher not use IV, just write empty method.

    public abstract void updateDecryptIV(byte[] iv);//If cipher not use IV, just write empty method.

    public abstract byte[] encrypt(byte[] data);

    public abstract byte[] decrypt(byte[] data);
}

File: jvmVer/src/main/java/com/proxy/shadowsocksr/impl/crypto/crypto/AbsCrypto.kt
package com.proxy.shadowsocksr.impl.crypto.crypto;

/*
 * If you need add new Crypto, just extends this class, and change CryptoManager class.
 *
 * If you need using C/C++ impl crypto, please in CryptoManager init it.
 */
abstract class AbsCrypto(val cryptMethod: String, val key: ByteArray)
{
    abstract fun updateEncryptIV(iv: ByteArray)
    abstract fun updateDecryptIV(iv: ByteArray)
    abstract fun encrypt(data: ByteArray): ByteArray
    abstract fun decrypt(data: ByteArray): ByteArray
}
